ObjectOrientedProgramming/MemoryOwnership/RAIICopySwapIdiom/src/dynamic_array.h

```diff
@@ -0,0 +1,41 @@
+#ifndef CPPBASICS_DYNAMIC_ARRAY_H
+#define CPPBASICS_DYNAMIC_ARRAY_H
+
+#include <iostream>
+
+class dynamic_array {
+private:
+    int *m_data = nullptr;
+    std::size_t m_size = 0;
+
+public:
+    dynamic_array();
+
+    explicit dynamic_array(std::size_t size, int init);
+
+    dynamic_array(const dynamic_array &other);
+
+    dynamic_array(dynamic_array &&other);
+
+    dynamic_array &operator=(const dynamic_array &other);
+
+    dynamic_array &operator=(dynamic_array &&other);
```

Comment:
please use "unifying assignment operator" idiom and leave just a single assignment operator overload:
https://en.wikibooks.org/wiki/More_C%2B%2B_Idioms/Copy-and-swap